compiler/control/OMROptions.cpp

```diff
@@ -701,7 +701,7 @@ TR::OptionTable OMR::Options::_jitOptions[] = {
    {"enableLateCleanFolding",            "O\tfold pdclean flags into pdstore nodes right before codegen",  SET_OPTION_BIT(TR_EnableLateCleanFolding), "F"},
    {"enableLinkagePreserveStrategy2",              "O\tenable linkage strategy 2", SET_OPTION_BIT(TR_LinkagePreserveStrategy2), "F"},
    {"enableLocalVPSkipLowFreqBlock",     "O\tSkip processing of low frequency blocks in localVP", SET_OPTION_BIT(TR_EnableLocalVPSkipLowFreqBlock), "F" },
-   {"enableLongRegAllocation",            "O\tenable allocation of 64-bit regs on 32-bit",      SET_OPTION_BIT(TR_Enable64BitRegsOn32Bit), "F"},
+   {"enableLongRegAllocation",            "O\tenable allocation of 64-bit regs on 32-bit",      SET_OPTION_BIT(TR_Enable64BitRegsOn32Bit), "T"},
```

Comment:
I don't think this does what you may think it does. Changing the `"F"` to `"T"` does not make this option enabled by default. i.e. `comp()->getOption(TR_Enable64BitRegsOn32Bit)` will return false. This doesn't sound like what the commit title says is happening here.

I'm a little worried here. How did you test your change? Did you hardcode somewhere to enable this option manually?

Reply:
When testing the change I hardcoded `return true` into `use64BitRegsOn32Bit` but before opening this pull request and running my final tests I thought this option should be respected so I tried to modify the default value of this option. It looks like this would be incorrect anyway as this option should only be enabled by default on z. Since I had previously hardcoded return true some of my tests were still valid and I did have this working at one point but I would like to fix this and run some more tests on my own before this is merged. 

How would you suggest settings this option, should it be done in `J9CodeGenerator.cpp`, `OMRCodeGenerator.cpp`, or should I hard code return true?

Reply:
I don't think the option is used on other platforms at all. You can confirm this. What I suggest we do is we rename the option to `TR_Disable64BitRegsOn32Bit` and change the code to be:

```
OMR::CodeGenerator::use64BitRegsOn32Bit()
   {
#ifdef TR_TARGET_S390
   if (TR::Compiler->target.is64Bit())
      return false;
   else
      {
      return !self()->comp()->getOption(TR_Disable64BitRegsOn32Bit);
      }
#endif // TR_TARGET_S390
   return false;
}
```

Then redo all your testing so we have a way to disable the feature if things go wrong. Please test by both enabling (which should be the default with your change) and disabling (by hardcoding the disable option in `jitPreProcess` for example) and running all testing. I'd like to make sure that when we enable this even if nothing breaks we can still disable the feature in the next little while to in case bugs arise down the testing line.

Reply:
It looks like a `TR_Disable64BitRegsOn32Bit` options already exists.

https://github.com/eclipse/omr/blob/master/compiler/control/OMROptions.cpp#L411

I don't think I can reuse this option since it is used in an optimization and might have unintended side effects if we were to later enable the disabling. Should a different but still similar name be used, such as `TR_Disable64BitRegistersOn32Bit`?

https://github.com/eclipse/omr/blob/master/compiler/optimizer/LocalOpts.cpp#L5984-L5994

Reply:
I think we should be fine to reuse the same option that already exists. If you check this `` optimization it's end result is deciding whether `useLongRegAllocation` will return true or false. This function is only used in `OMR::CodeGenerator::use64BitRegsOn32Bit()` - a use which you end up deleting as part of this PR. This effectively makes `TR_LongRegAllocation` dead code and any associated code in `TR_Rematerialization`. Sounds like there is more potential for cleanup. Do you want to take care of this in a new commit?

Reply:
Also please validate my logic here was correct in case I've missed something.

Reply:
Remove `TR_LongRegAllocation`: 752e355
Use `TR_Disable64BitRegsOn32Bit`: 31a08e1